src/main/java/uk/gov/cslearning/record/repository/BookingRepository.java

```diff
@@ -1,9 +1,15 @@
 package uk.gov.cslearning.record.repository;
 
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 import uk.gov.cslearning.record.domain.Booking;
+import uk.gov.cslearning.record.domain.Learner;
+
+import java.util.Optional;
 
 @Repository
 public interface BookingRepository extends CrudRepository<Booking, Integer>, CustomBookingRepository {
+    @Query("SELECT l FROM Booking b INNER JOIN Learner l ON b.learner = l INNER JOIN Event e ON b.event = e WHERE l.learnerEmail = ?1 AND e.uid = ?2")
+    Optional<Learner> findByLearnerEmailAndEventUid(String learnerEmail, String eventUid);
```

Comment:
This would look a little cleaner if the explicit joins were removed and you returned a booking rather than a learner (as this is the BookingRepository). Also you need to check the status of the booking.
```
@Query("select b from Booking where b.learner.learnerEmail = :email and b.event.uid = :eventUid and b.status in :status")
Optional<Booking> findByLearnerEmailAndEventUid(@Param("email") String learnerEmail, @Param("eventUid") String eventUid, @Param("status") List<String> status)
```